site/source/docs/tools_reference/settings_reference.rst

```diff
@@ -162,6 +170,9 @@ we need to copy the old heap into a new one in that case.
 If ALLOW_MEMORY_GROWTH is set, this initial amount of memory can increase
 later; if not, then it is the final and total amount of memory.
 
+By default, this value is calculated based on INITIAL_HEAP, STACK_SIZE,
+as well the size of static data in input modules.
```

Comment:
This is a bit hand-wavy, since it only actually "gets calculated" by the linker, but is conceptually correct.